src/main/java/recipeapplication/database/RecipeFirebaseCrud.java

```diff
@@ -22,21 +28,58 @@ public RecipeFirebaseCrud(Firestore firestore,
 
 	@Override
 	public void create(Recipe data) {
-		
-		this.firestore.collection(collection).document(data.getId()).set(data);
-		
+		System.out.println("In recipe creation section");
+		DocumentReference docRef = this.firestore.collection(collection).document(data.getId());
+		try {
+			if (!docRef.get().get().exists())  // Check if document already exists, to not allow overwriting it with create.
+				docRef.set(data);
+			else
+				throw new RuntimeException("Failed to create, Recipe with this ID already exists!");
+		} catch (InterruptedException | ExecutionException e) {
+			e.printStackTrace();
+		}
 	}
 
 	@Override
-	public void read(String document) {
-		// TODO Auto-generated method stub
-		
+	public Recipe read(String document) {
+		System.out.println("In recipe reading section");
+		try {
+			DocumentSnapshot docSnap = this.firestore.collection(collection).document(document).get().get();
+			if (!docSnap.exists())
+				throw new RuntimeException("Failed to read, recipe doesn't exist!");
+			else {
+				Map<String, Object> data = docSnap.getData();
+
+				return new Recipe(
```

Comment:
Check how @SagivAsraf created the object using an ObjectMapper 
https://github.com/ronenrozen/recipeServer/pull/8/files (UserFirebaseCrud)